src/conf_mode/system-logs.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2021 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from sys import exit
+
+from vyos import ConfigError
+from vyos import airbag
+from vyos.config import Config
+from vyos.logger import syslog
+from vyos.template import render
+from vyos.util import dict_search
+airbag.enable()
+
+# path to logrotate configs
+logrotate_atop_file = '/etc/logrotate.d/vyos-atop'
+logrotate_rsyslog_file = '/etc/logrotate.d/vyos-rsyslog'
+
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+
+    base = ['system', 'logs']
+    logs_config = conf.get_config_dict(base, key_mangling=('-', '_'))
+
+    return logs_config
+
+
+def verify(logs_config):
+    # Nothing to verify here
+    pass
+
+
+def generate(logs_config):
+    # get configuration for logrotate atop
+    logrotate_atop = dict_search('logs.logrotate.atop', logs_config)
+    # provide an empty dictionary if there is no config
+    if not logrotate_atop:
+        logrotate_atop = {}
+    # generate new config file for atop
+    syslog.debug('Adding logrotate config for atop')
+    render(logrotate_atop_file, 'logs/logrotate/vyos-atop.tmpl', logrotate_atop)
+
+    # get configuration for logrotate rsyslog
+    logrotate_rsyslog = dict_search('logs.logrotate.messages', logs_config)
+    # provide an empty dictionary if there is no config
+    if not logrotate_rsyslog:
```

Comment:
The common design always passes `logs_config` to the `render()` function with very very very few exceptions.

The code path:
```python
    # provide an empty dictionary if there is no config
    if not logrotate_rsyslog:
        logrotate_rsyslog= {}
```

is superfluous as this can and should be handled in the Jinja2 template.

Reply:
Possible if we pass the whole config received via `conf.get_config_dict` to a template. There will always be a dictionary in an output. We pass here to a template only a part of the dictionary that is necessary for this template and may not exist.
In this case, I wanted to keep templates simple. Should I move this logic from Python to Jinja?

Reply:
Please do so - then it is consistent wirh all other implementations.

Reply:
Was resolved by properly applying default values. Because of them, there always will be a dictionary with values.